node/metrics.go

```diff
@@ -40,22 +44,29 @@ type Metrics struct {
 	registry *prometheus.Registry
 	// socketAddr is the address at which the metrics server will be listening.
 	// should be in format ip:port
-	socketAddr string
+	socketAddr             string
+	operatorId             core.OperatorID
+	onchainMetricsInterval int64
+	tx                     core.Transactor
+	chainState             core.ChainState
 }
 
-func NewMetrics(eigenMetrics eigenmetrics.Metrics, reg *prometheus.Registry, logger logging.Logger, socketAddr string) *Metrics {
+func NewMetrics(eigenMetrics eigenmetrics.Metrics, reg *prometheus.Registry, logger logging.Logger, socketAddr string, operatorId core.OperatorID, onchainMetricsInterval int64, tx core.Transactor, chainState core.ChainState) *Metrics {
 
 	// Add Go module collectors
 	reg.MustRegister(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}))
 	reg.MustRegister(collectors.NewGoCollector())
 
 	metrics := &Metrics{
-		Registered: promauto.With(reg).NewGauge(
+		// The "type" label have values: stake_share, rank. The "stake_share" is stake share (in basis point),
+		// and the "rank" is operator's ranking (the operator with highest amount of stake ranked as 1) by stake share in the quorum.
+		Registered: promauto.With(reg).NewGaugeVec(
 			prometheus.GaugeOpts{
 				Namespace: Namespace,
 				Name:      "registered",
-				Help:      "indicator about if DA node is registered",
+				Help:      "the quorums the DA node is registered",
 			},
+			[]string{"quorum", "type"},
```

Comment:
wait I am very confused with this metrics. 
This says `Registered` which seems, operator is registered or not. So binary 0 and 1

but we are setting this metrics to stake_share and rank. I think it will create confusion and probably not a right use of this?

Reply:
instead I think the better way can be - have 3 diff metrics
1) stake_share and for that label would be `quorum` 
2) rank and for that label would be `quorum`
3) registered  for that label would be `quorum`

so basically you just slice by quorum for all 3 metrics.

Reply:
It means the stake registered for each quorum: one for the precentage info, the other for the rank by the stake
I'm not sure why it's confusing

Reply:
coz the metrics name is `registered` so this means- we are emitting the value of registered. but here the value is actually of a label and that's confusing.

Reply:
like for example
```
node_eigenda_blobs_total{quorum="0",type="number"} 97605
node_eigenda_blobs_total{quorum="0",type="size"} 1.16567530656e+11
```
the metrics value is of `node_eigenda_blobs_total` - and then you can slice by quorum and type to identify more params.

Reply:
now that I think of the above - even above is a bit wrong

Reply:
but if you think it's not confusing and will be clear to people directly using this metrics - we are okay. In case they use our dashboard, they would not have to worry about it. so I am okay

Reply:
You just have 2 time series per quorum: the stake share and rank. The same for node_eigenda_blobs_total.
There is no confusion point, no inefficiency as well.

Reply:
![image](https://github.com/Layr-Labs/eigenda/assets/21249320/08f94eca-ccec-4ccd-83a5-a834c3029d14)
from https://prometheus.io/docs/practices/naming/

Madhur makes good points and I feel it would be clearer if we move these labels into their own metrics time series.

Reply:
interesting, this may affect a lot of metrics we built

Reply:
yea I know. but I think we should at least start making sure new metrics are correct and eventually migrate other faulty ones.

Reply:
If you feel strongly about it, feel free to open a PR for it, I'll approve it